### ISSUE TITLE
C++14

### ISSUE THREAD

**Issue by tbisanz** [eudaq/eudaq#459]
Hey,

We'd love to build on lxplus with precompiled root versions from cmvfs - the issue is that the root builds have C++14 enabled, and (don't ask me why, I stopped asking) it won't work to build ROOT with C++14 and programs which link to ROOT/use ROOT dictionaries with a lower standard.

Could we have a flag `-DUSE_CXX14=ON` or so which will set:

https://github.com/eudaq/eudaq/blob/052d26b4516a11d8d222aa0318633f50032b9581/cmake/Platform.cmake#L4

instead of:
```
set (CMAKE_CXX_STANDARD 11)
set_property (GLOBAL PROPERTY CXX_STANDARD_REQUIRED ON)
```

```
if (USE_CXX14)
  set (CMAKE_CXX_STANDARD 14)
else()
  set (CMAKE_CXX_STANDARD 11)
endif()
set_property (GLOBAL PROPERTY CXX_STANDARD_REQUIRED ON)
```

Could we also backmerge that into 1.8?

**Comment by tbisanz** [comment]
By the way, the code will also enable GNU extensions, this might be a problem since you also build on OSX and with MSVS, right? If you don't want that:
```
set (CMAKE_CXX_EXTENSIONS OFF)
```
(or with `set_property`)

**Comment by dreyling** [comment]
@tbisanz: thanks! 

We (@eyiliu @jha92 and me) have defined now version 1.8 requires c++14 compiler. Thus, there is now an development branch: v1.8-dev 
And I have updated the v1.8.0 tag, which is used for the EUTelescope installation at the moment.

@arummler

**Comment by arummler** [comment]
I think it is better to have a v1x-trunk branch instead of always creating new branches. I had not done that yet as I have still some PRs pending.

**Comment by dreyling** [comment]
Ok, you mean one general `v1-dev` branch. How should we handle then in this example an upgrade requiring c++14?

**Comment by arummler** [comment]
Why the "backport" to tag 1.8.0 instead of tagging just 1.9.0? My release 1.8.0 changed now from release to draft and I am not sure how to fix it.

**Comment by dreyling** [comment]
Ok, I will do that -- after we have sort out that #441 
> Why the "backport" to tag 1.8.0 instead of tagging just 1.9.0?
Thus, we did the "backport".. 

But we can create of course another tag, which is then not chronological. 
@tbisanz What was the reason use the v1.8.0 tag for EUTelescope? Just to have the latest version or something else?

**Comment by dreyling** [comment]
@arummler ok, should be fixed now: tag 1.8.0 is as before. Can you please check: https://github.com/eudaq/eudaq/releases ?
In addition, tag 1.8.1 is just a copy, but requiring c++14 (for EUTelescope @tbisanz )

@arummler I also add a version 1 development branch as suggested: v1.x-dev
If you agree, we can delete branch v1.7-dev and close this issue, since v1.x-dev requires now also C++14.

**Comment by arummler** [comment]
I checked and it seems fine. All open PRs against v1.7-dev were made by myself and are either obsolete or I can reopen them later against the new branch. I deleted v1.7-dev and adapted the CI for the new branch.
